### PR TITLE
feat(ARCH-737/scripts): add jsx file extension support

### PR DESCRIPTION
**Pull request by jmfrancois**
**What is the problem this PR is trying to solve?**

We are in the process to make our code base support mainstream tools like vite.
We have identified vite by default wants  filename extension to be .jsx when the file has react syntax which make sens.

**What is the chosen solution to this problem?**

Add that to our talend-scripts default configuraiton

We have changed the playground to use jsx and it works !

**Please check if the PR fulfills these requirements**

- [x] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [x] Docs have been added / updated (for bug fixes / features)
- [x] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: .changeset/fuzzy-seas-stare.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+"@talend/scripts-config-react-webpack": minor
+"@talend/scripts-core": minor
```
Comment (smouillour) [inline .changeset/fuzzy-seas-stare.md:3]: There is no change in core
And there is changes in playwground

File: packages/playground/package.json [hunk 2]
```diff
@@ -9,6 +9,7 @@
     "test": "echo nothing to test in playground",
     "test:demo:umd": "cross-env BASENAME='/playground/' INITIATOR_URL='/playground/inject.js' talend-scripts build --prod",
     "start": "cross-env BASENAME='/' INITIATOR_URL='/inject.js' talend-scripts start --open http://localhost:3000",
+    "start:none": "talend-scripts start --open http://localhost:3000",
```
Comment (smouillour) [inline packages/playground/package.json:12]: I suppose we can remove this

File: .changeset/fuzzy-seas-stare.md (hunk 1 above)
Comment (jmfrancois) [inline .changeset/fuzzy-seas-stare.md:3]:
```suggestion
```

File: packages/playground/package.json (hunk 2 above)
Comment (jmfrancois) [inline packages/playground/package.json:12]:
```suggestion
```

**Review by smouillour — APPROVED** [review]

**Review by romainseb — APPROVED** [review]

**Review by tlnd-mhuchet — APPROVED** [review]

**Pull request merged**